src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/LibraryImportAttribute.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// Attribute used to indicate a source generator should create a function for marshaling
+    /// arguments instead of relying on the runtime to generate an equivalent marshaling function at run-time.
+    /// </summary>
+    /// <remarks>
+    /// This attribute is meaningless if the source generator associated with it is not enabled.
+    /// The associated source generator only supports C# and only supplies an implementation when
+    /// applied to static, partial, non-generic methods.
+    /// </remarks>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false)]
+#if SYSTEM_PRIVATE_CORELIB
+    public
+#else
+    internal
+#endif
+    sealed class LibraryImportAttribute : Attribute
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LibraryImportAttribute"/>.
+        /// </summary>
+        /// <param name="libraryName">Name of the library containing the import.</param>
+        public LibraryImportAttribute(string libraryName)
+        {
+            LibraryName = libraryName;
+        }
+
+        /// <summary>
+        /// Gets the name of the library containing the import.
+        /// </summary>
+        public string LibraryName { get; }
```

Comment:
Doesn't this need to be marked `init;`?

Reply:
I don't think we want `init`, since we only want to set it in the constructor, not any object initialization. For example, it would allow something like `new LibraryImportAttribute("Lib1") { LibraryName = "Lib2" }`, which I don't think we want.

It would mean also that everything including this would also need `IsExternalInit`.

Reply:
Thanks. That makes sense.